File: app/src/main/java/com/simplecity/amp_library/ui/dialog/ChangelogDialog.java
package com.simplecity.amp_library.ui.dialog;

import android.annotation.SuppressLint;
import android.content.Context;
import android.support.v4.content.ContextCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.CheckBox;
import android.widget.ProgressBar;

import com.afollestad.aesthetic.Aesthetic;
import com.afollestad.materialdialogs.MaterialDialog;
import com.simplecity.amp_library.R;
import com.simplecity.amp_library.utils.DialogUtils;
import com.simplecity.amp_library.utils.SettingsManager;
import com.simplecity.amp_library.utils.ViewUtils;

public class ChangelogDialog {

    private ChangelogDialog() {
        //no instance
    }

    public static MaterialDialog getChangelogDialog(Context context) {

        @SuppressLint("InflateParams")
        View customView = LayoutInflater.from(context).inflate(R.layout.dialog_changelog, null);

        WebView webView = customView.findViewById(R.id.webView);
        webView.setBackgroundColor(ContextCompat.getColor(context, android.R.color.transparent));

        CheckBox checkBox = customView.findViewById(R.id.checkbox);
        checkBox.setChecked(SettingsManager.getInstance().getShowChangelogOnLaunch());
        checkBox.setOnCheckedChangeListener((buttonView, isChecked) -> SettingsManager.getInstance().setShowChangelogOnLaunch(isChecked));

        ProgressBar progressBar = customView.findViewById(R.id.progress);

        webView.setWebViewClient(new WebViewClient() {
            @Override
            public void onPageFinished(WebView view, String url) {
                super.onPageFinished(view, url);

                ViewUtils.fadeOut(progressBar, ()
                        -> ViewUtils.fadeIn(webView, null));
            }
        });

        Aesthetic.get(context)
                .isDark()
                .take(1)
                .subscribe(isDark -> webView.loadUrl(isDark ? "file:///android_asset/web/info_dark.html" : "file:///android_asset/web/info.html"));

        return DialogUtils.getBuilder(context)
                .title(R.string.pref_title_changelog)
                .customView(customView, false)
                .negativeText(R.string.close)
                .build();
    }
}

File: app/src/main/java/com/simplecity/amp_library/ui/dialog/ChangelogDialog.kt
package com.simplecity.amp_library.ui.dialog

import android.annotation.SuppressLint
import android.app.Dialog
import android.content.Context
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.support.v4.app.FragmentManager
import android.support.v4.content.ContextCompat
import android.view.LayoutInflater
import android.webkit.WebView
import android.webkit.WebViewClient
import android.widget.CheckBox
import android.widget.ProgressBar
import com.afollestad.aesthetic.Aesthetic
import com.afollestad.materialdialogs.MaterialDialog
import com.simplecity.amp_library.R
import com.simplecity.amp_library.utils.SettingsManager
import com.simplecity.amp_library.utils.ViewUtils
import dagger.android.support.AndroidSupportInjection
import javax.inject.Inject

class ChangelogDialog : DialogFragment() {

    @Inject lateinit var settingsManager: SettingsManager

    override fun onAttach(context: Context?) {
        AndroidSupportInjection.inject(this)
        super.onAttach(context)
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        @SuppressLint("InflateParams")
        val customView = LayoutInflater.from(context).inflate(R.layout.dialog_changelog, null)

        val webView = customView.findViewById<WebView>(R.id.webView)
        webView.setBackgroundColor(ContextCompat.getColor(context!!, android.R.color.transparent))

        val checkBox = customView.findViewById<CheckBox>(R.id.checkbox)
        checkBox.isChecked = settingsManager.showChangelogOnLaunch
        checkBox.setOnCheckedChangeListener { buttonView, isChecked -> settingsManager.showChangelogOnLaunch = isChecked }

        val progressBar = customView.findViewById<ProgressBar>(R.id.progress)

        webView.webViewClient = object : WebViewClient() {
            override fun onPageFinished(view: WebView, url: String) {
                super.onPageFinished(view, url)

                ViewUtils.fadeOut(progressBar) { ViewUtils.fadeIn(webView, null) }
            }

            override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
                val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
                if (intent.resolveActivity(context!!.packageManager) != null) {
                    context?.startActivity(intent)
                    return true
                }
                return false
            }
        }

        Aesthetic.get(context)
            .isDark
            .take(1)
            .subscribe { isDark -> webView.loadUrl(if (isDark) "file:///android_asset/web/info_dark.html" else "file:///android_asset/web/info.html") }

        return MaterialDialog.Builder(context!!)
            .title(R.string.pref_title_changelog)
            .customView(customView, false)
            .negativeText(R.string.close)
            .build()
    }

    fun show(fragmentManager: FragmentManager) {
        show(fragmentManager, TAG)
    }

    companion object {
        private const val TAG = "ChangelogDialog"

        fun newInstance() = ChangelogDialog()
    }
}